File: src/main/java/de/sean/splugin/spigot/tasks/AfkChecker.java
package de.sean.splugin.spigot.tasks;

/* SPlugin */
import de.sean.splugin.util.SMessages;
import de.sean.splugin.util.SUtil;

/* Spigot */
import org.bukkit.Bukkit;
import org.bukkit.entity.Player;

public class AfkChecker implements Runnable {
    @Override
    public void run() {
        Bukkit.getOnlinePlayers()
            .forEach(this::checkPlayer);
    }

    private void checkPlayer(final Player player) {
        final long lastActivity = SUtil.getLastActivityForPlayer(player.getUniqueId());
        final long curTime = System.currentTimeMillis();

        // 120000 Milliseconds = 2 Minutes
        if ((curTime - lastActivity) > 120000 && !SUtil.isPlayerAFK(player.getUniqueId())) {
            SUtil.setPlayerAFK(player.getUniqueId(), true);
            SMessages.markPlayerAFK(player);
        }
    }
}

File: src/main/kotlin/de/sean/splugin/spigot/tasks/AfkChecker.kt
package de.sean.splugin.spigot.tasks

import de.sean.splugin.util.SMessages.markPlayerAFK
import de.sean.splugin.util.SUtil
import org.bukkit.Bukkit
import org.bukkit.entity.Player

class AfkChecker : Runnable {
    override fun run() {
        Bukkit.getOnlinePlayers().forEach { player: Player -> checkPlayer(player) }
    }

    private fun checkPlayer(player: Player) {
        val lastActivity = SUtil.playerLastActivity[player.uniqueId]
        val curTime = System.currentTimeMillis()

        // 120000 Milliseconds = 2 Minutes
        if (curTime - lastActivity!! > 120000 && !SUtil.afkPlayers[player.uniqueId]!!) {
            SUtil.afkPlayers[player.uniqueId] = true
            markPlayerAFK(player)
        }
    }
}
